src/SecTester.Scan/Scan.cs

```diff
@@ -96,6 +96,45 @@ public async Task Stop()
     yield return _scanState;
   }
 
+  public async Task Expect(Func<Scan, Task<bool>> predicate)
+  {
+    if (predicate == null)
+    {
+      throw new ArgumentNullException(nameof(predicate));
+    }
+
+    using var cancellationTokenSource = new CancellationTokenSource(this._options.Timeout);
+
+    await Expect(predicate, cancellationTokenSource.Token).ConfigureAwait(false);
+  }
+
+  private async Task<bool> SatisfyPredicate(Func<Scan, Task<bool>> predicate)
```

Comment:
This is quite a weird name. To say if your argument is satisfied predicated, you have to apply it to the argument first.
```suggestion
  private async Task<bool> ApplyPredicate(Func<Scan, Task<bool>> predicate)
```